<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogConnectionDetails</class>
 <widget class="QDialog" name="DialogConnectionDetails">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>539</width>
    <height>406</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Connection Details</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="leftMargin">
      <number>30</number>
     </property>
     <item>
      <widget class="QLabel" name="labelConnectUsing">
       <property name="text">
        <string>Connect Using:</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="ConnectionComboBox" name="comboBoxConnectUsing">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonExcludeVirtualPorts">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Exclude Virtual Serial Ports</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../resources.qrc">
           <normaloff>:/res/serialPortFilter.png</normaloff>:/res/serialPortFilter.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>100</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout">
         <property name="labelAlignment">
          <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="labelIPAddress">
           <property name="text">
            <string>IP Address:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineEditIPAddress">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string notr="true">127.0.0.1</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelServicePort">
           <property name="text">
            <string>Service Port:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="NumericLineEdit" name="lineEditServicePort">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>25</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string notr="true">502</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="horizontalSpacer_8">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>120</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Configuration</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <property name="spacing">
       <number>30</number>
      </property>
      <property name="leftMargin">
       <number>18</number>
      </property>
      <item>
       <layout class="QFormLayout" name="formLayout_2">
        <property name="labelAlignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="formAlignment">
         <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="horizontalSpacing">
         <number>3</number>
        </property>
        <property name="verticalSpacing">
         <number>6</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="labelBaudRate">
          <property name="text">
           <string>Baud Rate: </string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="NumericComboBox" name="comboBoxBaudRate">
          <property name="minimumSize">
           <size>
            <width>90</width>
            <height>25</height>
           </size>
          </property>
          <property name="editable">
           <bool>true</bool>
          </property>
          <property name="currentIndex">
           <number>3</number>
          </property>
          <item>
           <property name="text">
            <string notr="true">1200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">2400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">4800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">9600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">19200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">38400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">57600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">115200</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelWordLength">
          <property name="text">
           <string>Word Length: </string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="NumericComboBox" name="comboBoxWordLength">
          <property name="minimumSize">
           <size>
            <width>90</width>
            <height>25</height>
           </size>
          </property>
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string notr="true">7</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">8</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="labelParity">
          <property name="text">
           <string>Parity: </string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="ParityTypeComboBox" name="comboBoxParity">
          <property name="minimumSize">
           <size>
            <width>90</width>
            <height>25</height>
           </size>
          </property>
          <property name="currentIndex">
           <number>-1</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Stop Bits: </string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="NumericComboBox" name="comboBoxStopBits">
          <property name="minimumSize">
           <size>
            <width>90</width>
            <height>25</height>
           </size>
          </property>
          <item>
           <property name="text">
            <string notr="true">1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">2</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Hardware Flow Control</string>
        </property>
        <layout class="QFormLayout" name="formLayout_3">
         <property name="labelAlignment">
          <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="labelFlowControl">
           <property name="text">
            <string>Flow Control Mode: </string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="FlowControlTypeComboBox" name="comboBoxFlowControl">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>25</height>
            </size>
           </property>
           <property name="currentIndex">
            <number>-1</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="labelDTRControl">
           <property name="text">
            <string>DTR Control: </string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="BooleanComboBox" name="comboBoxDTRControl">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>90</width>
             <height>25</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="labelRTSControl">
           <property name="text">
            <string>RTS Control: </string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="BooleanComboBox" name="comboBoxRTSControl">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>90</width>
             <height>25</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <spacer name="horizontalSpacer_9">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonProtocolSelections">
       <property name="minimumSize">
        <size>
         <width>110</width>
         <height>26</height>
        </size>
       </property>
       <property name="text">
        <string>Protocol Selections</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_10">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NumericLineEdit</class>
   <extends>QLineEdit</extends>
   <header>numericlineedit.h</header>
  </customwidget>
  <customwidget>
   <class>NumericComboBox</class>
   <extends>QComboBox</extends>
   <header>numericcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>ConnectionComboBox</class>
   <extends>QComboBox</extends>
   <header>connectioncombobox.h</header>
  </customwidget>
  <customwidget>
   <class>ParityTypeComboBox</class>
   <extends>QComboBox</extends>
   <header>paritytypecombobox.h</header>
  </customwidget>
  <customwidget>
   <class>FlowControlTypeComboBox</class>
   <extends>QComboBox</extends>
   <header>flowcontroltypecombobox.h</header>
  </customwidget>
  <customwidget>
   <class>BooleanComboBox</class>
   <extends>QComboBox</extends>
   <header>booleancombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogConnectionDetails</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogConnectionDetails</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
